src/test/java/teammates/test/cases/browsertests/FeedbackMcqQuestionUiTest.java

```diff
@@ -291,4 +294,294 @@ public void testDeleteQuestionAction() {
+        feedbackEditPage.fillQuestionDescription("More details", NEW_QUESTION_INDEX);
+        feedbackEditPage.clickMcqAssignWeightCheckboxForNewQuestion();
+        feedbackEditPage.clickAddMcqOtherOptionCheckboxForNewQuestion();
+
+        ______TS("MCQ: Test that selecting generated option hides mcq weights cells and checkbox");
+
+        // Test visibility of mcq weights and checkbox before selecting mcq generated option.
+        assertTrue(feedbackEditPage.isElementVisible(By.id("mcqAssignWeights--1")));
+        assertTrue(feedbackEditPage.isElementVisible(By.id("mcqWeights--1")));
+        assertTrue(feedbackEditPage.isElementVisible(By.id("mcqOtherWeight--1")));
+
+        feedbackEditPage.clickGenerateMcqOptionsCheckbox(NEW_QUESTION_INDEX);
+
+        // Check that 'choices are weighted' checkbox is hidden
+        assertFalse(feedbackEditPage.isElementVisible(By.id("mcqAssignWeights--1")));
+        assertFalse(feedbackEditPage.isElementVisible(By.id("mcqWeights--1")));
+        assertFalse(feedbackEditPage.isElementVisible(By.id("mcqOtherWeight--1")));
+
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_ADDED);
+
+        ______TS("MCQ: Test mcq weight have default values after edit to generated options");
+        feedbackEditPage.clickEditQuestionButton(1);
+        feedbackEditPage.clickGenerateMcqOptionsCheckbox(1); // Uncheck the generated option checkbox
+
+        // Check that weight fields are again visible after unchecking generated option,
+        // and also check that all the fields have the default values stored.
+        assertFalse(feedbackEditPage.isMcqAssignWeightCheckboxChecked(1));
+        assertFalse(browser.driver.findElement(By.id("mcqOtherOptionFlag-1")).isSelected());
+
+        // Add One option and check MCQ assign weight checkbox to check the mcq weight field value.
+        feedbackEditPage.clickAddMoreMcqOptionLink(1);
+        feedbackEditPage.clickMcqAssignWeightsCheckbox(1);
+        assertTrue(feedbackEditPage.isElementVisible("mcqWeight-0-1"));
+        feedbackEditPage.verifyFieldValue("mcqWeight-0-1", "0");
+
+        // Check the other option to test the value of other weight
+        feedbackEditPage.clickAddMcqOtherOptionCheckbox(1);
+        assertTrue(feedbackEditPage.isElementVisible(By.id("mcqOtherWeight-1")));
+        feedbackEditPage.verifyFieldValue("mcqOtherWeight-1", "0");
+
+        feedbackEditPage.clickDeleteQuestionLink(1);
+        feedbackEditPage.waitForConfirmationModalAndClickOk();
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_DELETED);
+        assertNull(BackDoor.getFeedbackQuestion(courseId, feedbackSessionName, 1));
+    }
+
+    @Test(priority = 1)
+    public void testMcqWeightsFeature_instructorResultsPageQuestionView_showStatistics() throws Exception {
```

Comment:
I believe this test should be `InstructorFeedbackResultsPageUiTest` here?

Reply:
I also thought that initially, but decided against it when I saw Rubric questions also have the results page tests in `FeedbackRubricQuestionUiTest` file, and one other reason was that I couldn't find any specific question related test in `InstructorFeedbackResultsPageUiTest`, all the tests seems more generic instead of focusing on a single aspect of a single question.

So should I move this test in `InstructorFeedbackResultsPageUiTest`? Wouldn't it be better if specific question related tests belong to a particular file?

Reply:
Sure, cat put it here. Remember to follow the convention in your later PRs.